src/node/src/storage_node_impl.rs

```diff
@@ -164,21 +179,69 @@ impl StorageNode for StorageNodeImpl {
                         )));
                     }
                 }
-
-                // Submit query session
-
                 // Takes a reference and returns Option<&V>
                 let sess = sess_store
                     .remove_session(&payload.session_token)
                     .map_err(|e| Status::internal(format!("{}", e)))
                     .unwrap();
-                // TODO(chenjing): sign
-                Ok(Response::new(CloseSessionResponse {
-                    query_session_info: Some(sess.get_session_info()),
-                }))
+                node_query_session_info = Some(sess.get_session_info());
             }
-            Err(e) => Err(Status::internal(format!("{}", e))),
+            Err(e) => return Err(Status::internal(format!("{}", e))),
         }
+        // Generate Nonce
+        let nonce = match SystemTime::now().duration_since(UNIX_EPOCH) {
```

Comment:
Borrow nonce logic from cmd/lib.rs
Will it be affected by 
https://github.com/dbpunk-labs/db3/issues/190 ?